sympy/ntheory/factor_.py

```diff
@@ -420,12 +420,26 @@ def perfect_power(n, candidates=None, big=True, factor=True):
     Examples
     ========
 
-    >>> from sympy import perfect_power
+    >>> from sympy import perfect_power, Rational
     >>> perfect_power(16)
     (2, 4)
     >>> perfect_power(16, big=False)
     (4, 2)
 
+    Negative numbers can only have odd perfect powers:
+
+    >>> perfect_power(-4)
+    False
+    >>> perfect_power(-8)
+    (-2, 3)
+
+    Rationals are also recognized:
+
+    >>> perfect_power(Rational(1/2)**3)
+    (1/2, 3)
+    >>> perfect_power(Rational(-3/2)**3)
```

Comment:
Should this be e.g. `Rational(-3, 2)`?

Reply:
denominators that are powers of 2 are exactly converted...but it is better practice to do as you show
```python
>>> Rational(1/4)
1/4
```